crates/ruff/src/rules/dlint/rules/bad_compile_use.rs

```diff
@@ -6,6 +6,22 @@ use ruff_macros::{derive_message_formats, violation};
 
 use crate::checkers::ast::Checker;
 
+/// ## What it does
+/// Checks that code does not call the built-in function `compile`
+///
+/// ## Why is this bad?
+/// While not bad in and of itself, this function is _probably_ a code smell that
+/// something else we don't want could be going on. I.e. "compile" is often proceeded by
+/// "eval" or "exec".
+///
+/// ## Example
+/// ```python
+/// compile("foo")
+/// ```
+///
+/// Use instead:
+/// ```python
+/// ```
 #[violation]
 pub struct BadCompileUse;
```

Comment:
I think we should rename the rule to simply `compile`, `eval` etc. because the bad seems redundant, all lint rules capture something "bad". This ensures that the rule reads nicely in the `allow compile` or `deny compile` context

Reply:
Names are straight copies from DLint. Happy to deviate from those if alright for straight plugin ports.

Reply:
Only problem is there is already an Eval rule in `rules::pygrep_hooks::rules::Eval`. For consistency's sake: EvalUse, ExecUse, etc.?

Reply:
I agree with Micha on the rule renaming.

Though think the biggest question with this PR is that most of these rules are already implemented under different names and categorizations: `rules::pygrep_hooks::rules::Eval`, `rules::flake8_bandit::rules::ExecBuiltin`, and then the various rules in `rules/flake8_bandit/rules/suspicious_function_call.rs` which cover `marshal` and `shelve`. I don't think we have a rule for `compile`, but otherwise, these are already mostly covered, so I'm hesitant to add duplicate implementations of the same violation enforcement.

Reply:
Sounds good, should probably have done that before diving in. Have asked requester if they already have one or more rules they use from DLint not covered by other plugins in Ruff. Don't think I have time to cross reference the issue lists for the next few days.

Reply:
I wish I'd noticed it and called it out earlier -- sorry to only do that now. It's hard for me to keep up with and proactively vet issues like that before they make it to PR.

Reply:
No worries. It's a busy repo, I learned a lot from @MichaReiser 's very thorough review (thanks for that!) about Rust and Ruff which I can bring with me to future edits even if this never gets to main.